radixdlt-core/radixdlt/src/main/java/com/radixdlt/statecomputer/substatehash/BerkeleySubstateAccumulatorHashStore.java

```diff
@@ -211,8 +215,25 @@ public void process(
     }
   }
 
-  private void persistCurrentSubstateAccumulatorHash(Transaction dbTxn) {
-    var key = new DatabaseEntry(SUBSTATE_ACCUMULATOR_HASH_KEY);
+  private void assertLastStateVersionIsAsExpected(long stateVersion) {
+    long expectedStateVersion = stateVersion - 1;
+    if (this.lastStateVersionInDbOpt.orElse(0L) != expectedStateVersion) {
+      throw new IllegalStateException(
+          String.format(
+              "The substate hash accumulator state version has got out of sync with the ledger (It"
+                  + " is expected to be at %s, but is at %s) - please clean the ledger and start"
+                  + " again.",
+              expectedStateVersion, this.lastStateVersionInDbOpt));
+    }
+  }
+
+  private void persistCurrentSubstateAccumulatorHash(
+      Transaction dbTxn, Optional<Long> currentStateVersion, long nextStateVersion) {
+    currentStateVersion.ifPresent(
+        it ->
```

Comment:
I'd prefer a more informative name here.

Reply:
Repeating `currentStateVersion`? I can change it, but seems unnecessary noise in this case. What do you think?

Reply:
Actually, I can't use the same name as it is already defined in the scope. I'll merge it like that for now. Please tell me what you suggest and I can change it in the next PR.